docs/design/908-proposal-process.md

```diff
@@ -0,0 +1,410 @@
+may be performed asynchronously or synchronously, up to the discretion of the
+governance members.
+
+The goal of proposal review is to make sure proposals are moving forward and
+are receiving attention from the right people. Governance members participating
+in proposal review should cc relevant maintainers, raise important questions,
+ping on stale discussions, and generally drive discussion towards some kind of
+consensus.
+
+The proposal review process also identifies when consensus has been reached on
+a proposal and moves that proposal to the next stage.
+
+The Alloy governance team can, at their discretion, make exceptions that do not
+need to go through all stages, fast-tracking them to any other stage. This may
+be appropriate for proposals that do not merit the full review process or
+proposals that need to be considered quickly, such as an upcoming release
+deadline.
+
+Proposals are in one of the following states:
+
+* **Incoming**: New proposals are given the _Incoming_ state.
+
+  The proposal review process prioritizes proposals in other states to keep
+  things moving forward. If time permits during the review process, proposals
+  in _Incoming_ are moved to _Active_.
+
+* **Active**: Proposals in the _Active_ state are reviewed during the proposal
+  review process to watch for emerging consensus in the discussion. The
+  proposal review group may also comment, make suggestions, ask clarifying
+  questions, and try to restate the proposal to make sure everyone agrees about
+  what exactly is being discussed.
+
+* **Likely Accept**: If a proposal discussion seems to have reached a consensus
+  to accept a proposal, the proposal review group moves the proposal to the
+  _Likely Accept_ state. This change should be indicated with a comment on the
+  proposal. This state marks the final period for comments that might change
+  the recognition of consensus.
+
+* **Likely Decline**: If a proposal discussion seems to have reached a consensus
+  to decline a proposal, the proposal review group moves the proposal to the
+  _Likely Decline_ state.
+
+  A proposal may also be moved to _Likely Decline_ if the proposal review group
+  identifies that no consensus is likely to be reached and that the default of
+  not accepting the proposal is appropriate.
+
+  Similarly to _Likely Accept_, this change should be indicated with a comment
+  on the proposal, and this state marks the final period for comments that
+  might change the recognition of consensus.
+
+* **Accepted**: If a proposal has been in _Likely Accept_ for a week and
+  consensus has not changed, the proposal is moved to the _Accepted_ state.
+
+  If significant discussion happens during the _Likely Accept_ period, the
+  proposal review group may decide to leave the proposal in _Likely Accept_ for
+  another week or even move the proposal back to _Active_.
+
+  Once a proposal has been Accepted, implementation work may begin.
```

Comment:
What do we do if we accept one proposal, but then circumstances change or a new proposal comes around and we want to decline a previously accepted proposal?

Reply:
I don't think "Accepted" means that the accepted solution will be accepted forever; software will continue to evolve over time. I think it's fine if a new proposal supersedes a previously accepted one; that doesn't mean the older proposal becomes rejected. 

I don't anticipate superseding proposals to be frequent, or to happen shortly after the superseded proposal gets accepted. If someone does manage to supersede a previously accepted proposal shortly after that proposal was accepted, I believe that points to a failing in the consensus process where the conversation failed to identify a better solution.

Reply:
Sounds good!